includes/subscriptions/class-wc-payments-subscriptions-migrator.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * Class WC_Payments_Subscriptions_Migrator
+ *
+ * @package WooCommerce\Payments
+ */
+
+use WCPay\Exceptions\API_Exception;
+
+/**
+ * Handles migrating WCPay Subscriptions to tokenized subscriptions.
+ */
+class WC_Payments_Subscriptions_Migrator {
+
+	/**
+	 * Valid subscription statuses to cancel a subscription at Stripe.
+	 *
+	 * @var array $active_statuses
+	 */
+	private $active_statuses = [ 'active', 'past_due', 'trialing', 'paused' ];
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// Hook onto Scheduled Action to migrate wcpay subscription.
+		// add_action( 'wcpay_migrate_subscription', [ $this, 'migrate_wcpay_subscription' ] );.
+	}
+
+	/**
+	 * Migrate WCPay Subscription to WC Subscriptions
+	 *
+	 * Migration process:
+	 * 1. Validate the request to migrate subscription
+	 * 2. Fetches the subscription from Stripe
+	 * 3. Cancels the subscription at stripe if it is active
+	 * 4. Update the subscription meta to indicate that it has been migrated
+	 * 5. Add an order note on the subscription
+	 *
+	 * @param int $subscription_id The ID of the subscription to migrate.
+	 */
+	public function migrate_wcpay_subscription( $subscription_id ) {
+		$subscription = $this->fetch_and_validate_subscription_to_migrate( $subscription_id );
```

Comment:
I think going with a try/catch approach would clean up this migration function a lot because when I originally implemented this, it was all in one function. Then I decided to separate it out into individual functions but the end result was having to return true/false and add a bunch of conditions to exit out of the migration early with a specific log.

I'll refactor this file to use a try/catch approach and see how that looks.

Reply:
I've updated the migrate function to surround the code with a try-catch block in this commit: [0e0bffa](https://github.com/Automattic/woocommerce-payments/pull/6706/commits/0e0bffa595dccf13806ea310a8d6fb429abb5899)

This feels a bit cleaner than the original approach to me, but I'm keen to see if you think the same.